### PR TITLE
configs to support CP2K QM/MM in GROMACS

### PR DESCRIPTION
**Pull request by ostueker**
easyconfigs needed to evaluate changes to GROMACS easyblock changes in easybuilders/easybuild-easyblocks#2750

* GROMACS-2022.3-foss-2020a-CP2K.eb 
* CP2K-2022.1-foss-2020a.eb

### CODE REVIEW

**Comment by ostueker** [comment]
The error in the tests (see below) is expected, because the variable `mpi_only` has been introduced in easybuilders/easybuild-easyblocks#2750 [here](https://github.com/easybuilders/easybuild-easyblocks/pull/2750/files#diff-528f8ee1c26eeeb855e4beced90381894b8bbb583cba1f8545b97e9fe13a24abR69) .


```
======================================================================
ERROR: test__parse_easyconfig_GROMACS-2022.3-foss-2020a-CP2K.eb (test.easyconfigs.easyconfigs.EasyConfigTest)
Test for easyconfig GROMACS-2022.3-foss-2020a-CP2K.eb
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/easyconfigs/easyconfigs.py", line 1481, in innertest
    template_easyconfig_test(self, spec_path)
  File "test/easyconfigs/easyconfigs.py", line 1202, in template_easyconfig_test
    ecs = process_easyconfig(spec)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/easybuild/framework/easyconfig/easyconfig.py", line 2071, in process_easyconfig
    raise EasyBuildError("Failed to process easyconfig %s: %s", spec, err.msg)
EasyBuildError: "Failed to process easyconfig /home/runner/work/easybuild-easyconfigs/easybuild-easyconfigs/easybuild/easyconfigs/g/GROMACS/GROMACS-2022.3-foss-2020a-CP2K.eb: Use of 1 unknown easyconfig parameters detected in GROMACS-2022.3-foss-2020a-CP2K.eb: mpi_only\nIf these are just local variables please rename them to start with 'local_', or try using --fix-deprecated-easyconfigs to do this automatically.\nFor more information, see https://easybuild.readthedocs.io/en/latest/Easyconfig-files-local-variables.html ."
```

**Comment by ostueker** [comment]
The error is a :chicken: and :egg:  problem:

* this PR depends on changes implemented in PR easybuilders/easybuild-easyblocks#2750
* and PR easybuilders/easybuild-easyblocks#2750 won't get merged without having been tested with these easyconfigs.

File: easybuild/easyconfigs/c/CP2K/CP2K-2022.1-foss-2020a.eb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# get FFTW from toolchain
+dependencies = [
+    ('Libint', '2.6.0', '-lmax-6-cp2k'),
+    ('libxc', '5.1.3'),
+    ('libxsmm', '1.16.1'),
+]
+
+builddependencies = [
+    ('flex', '2.6.4'),
+    ('Bison', '3.7.1'),
```
Comment (akesandgren) [inline easybuild/easyconfigs/c/CP2K/CP2K-2022.1-foss-2020a.eb:26]: For 2020a toolchain use Bison 3.5.3

File: easybuild/easyconfigs/c/CP2K/CP2K-2022.1-foss-2020a.eb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+toolchainopts = {'pic': True, 'openmp': True}
+
+source_urls = ['https://github.com/cp2k/cp2k/releases/download/v%(version)s.0/']
+sources = [SOURCELOWER_TAR_BZ2]
+checksums = ['2c34f1a7972973c62d471cd35856f444f11ab22f2ff930f6ead20f3454fd228b']
+
+# get FFTW from toolchain
+dependencies = [
+    ('Libint', '2.6.0', '-lmax-6-cp2k'),
+    ('libxc', '5.1.3'),
```
Comment (akesandgren) [inline easybuild/easyconfigs/c/CP2K/CP2K-2022.1-foss-2020a.eb:20]: For 2020a toolchain use libxc 4.3.4

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2022.3-foss-2020a-CP2K.eb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    'ftp://ftp.gromacs.org/pub/gromacs/',
+]
+sources = [SOURCELOWER_TAR_GZ]
+patches = []
+checksums = [
+    '14cfb130ddaf8f759a3af643c04f5a0d0d32b09bc3448b16afa5b617f5e35dae',  # gromacs-2022.3.tar.gz
+]
+
+builddependencies = [
+    ('CMake', '3.23.1'),
```
Comment (akesandgren) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2022.3-foss-2020a-CP2K.eb:49]: For 2020a toolchain use CMake 3.16.4

**Review by akesandgren — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
closing this since `foss/2020a` is deprecated, see https://docs.easybuild.io/policies/toolchains

**Pull request closed without merging**